mat64/inner.go

```diff
@@ -29,26 +29,65 @@ func Inner(x []float64, A Matrix, y []float64) float64 {
 	switch b := A.(type) {
 	case RawSymmetricer:
 		bmat := b.RawSymmetric()
```

Comment:
This should probably be "case RawSymmetricer, RawTriangular", and then get the data slice and stride as variables. Also, we should check that the Uplo is correct.